tests/test_project.py

```diff
@@ -417,6 +417,65 @@ def updater(remotes):
     assert ur.tr_head_0 == v1_0
     assert ur.tr_head_1 == v2_0
 
+def test_update_head_0(west_init_tmpdir):
+    # Verify that using HEAD~0 as the revision causes west to not touch the
+    # local repo. In this case, local zephyr should remain unchanged, even
+    # though it is referenced in the new top-level manifest 'my_manifest',
+    # since it's referenced with HEAD~0. Check that a local commit remains, and
+    # that local changes are kept as is. An invalid url ('na') is used to check
+    # that west doesn't attempt to access it.
+    # Expect only Kconfiglib to be updated, since it's mentioned in the new
+    # manifest's name-allowlist.
+
+    # create local repositories
+    cmd('update')
+
+    local_zephyr = west_init_tmpdir / "zephyr"
+
+    add_commit(local_zephyr, "local commit")
+    local_commit = check_output([GIT, 'rev-parse', 'HEAD'], cwd=local_zephyr)
```

Comment:
For consistency and clarity?
```suggestion
    local_commit = check_output([GIT, 'rev-parse', 'HEAD~0'], cwd=local_zephyr)
```

Reply:
I think in this case we really want to see HEAD. The `HEAD~0` trick is what a test case is being added for, but if there's no need for the extra verbosity in other places (there isn't) we may as well omit it.